src/rust/rust-proto-new/src/serde_impl.rs

```diff
@@ -0,0 +1,36 @@
+use bytes::{
+    Bytes,
+    BytesMut,
+};
+use prost::Message;
+
+use crate::{
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+pub trait HasAssociatedProto<T> {
+    type Proto: From<T> + Message + Default + TryInto<T, Error = SerDeError>;
```

Comment:
Why is `Default` here?

Reply:
It's needed for Message::decode, which kinda makes sense (proto deserialization usually constructs an empty/default message and then mutates it)